src/main/java/core/reader/TabDelimitedReader.java

```diff
@@ -0,0 +1,203 @@
+package core.reader;
+
+import static core.account.AccountSide.CREDIT;
+import static core.account.AccountSide.DEBIT;
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+
+import core.Journal;
+import core.Ledger;
+import core.account.AccountDetails;
+import core.account.AccountDetailsImpl;
+import core.account.AccountSide;
+import core.account.AccountingEntry;
+import core.chartofaccounts.ChartOfAccounts;
+import core.transaction.AccountingTransaction;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import lombok.Value;
+
+/**
+ A TabDelimitedReader reads the chart of accounts and transactions
+ from a tab-delimited file.
+
+ <p>
+   See sample files in test/resources for examples.
+   Skips blanks lines and lines that start with a '#'.
+ </p>
+
+ */
+@Value
+public class TabDelimitedReader implements Reader {
+
```

Comment:
Please format the code in a way that is consistent with the existing code, e. g. no new lines at the beginning or end of class or methods, no double new lines, etc.

Reply:
Do you by any chance have a style def I can import into my editor (IntelliJ)?

Reply:
I used the IntelliJ default style settings.